gpytorch/kernels/distributional_input_kernel.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+
+from torch.distributions import Normal, kl_divergence
+
+from .kernel import Kernel
+
+
+def _symmetrized_kl(dist1, dist2, num_dims=2, eps=1e-8):
+    dist1_mean = dist1[..., :num_dims].unsqueeze(-3)
+    dist1_logvar = dist1[..., num_dims:].unsqueeze(-2)
+    dist1_var = eps + dist1_logvar.exp()
+
+    dist2_mean = dist2[..., :num_dims].unsqueeze(-3)
+    dist2_logvar = dist2[..., num_dims:].unsqueeze(-2)
+    dist2_var = eps + dist2_logvar.exp()
+
+    dist1 = Normal(dist1_mean, dist1_var ** 0.5)
+    dist2 = Normal(dist2_mean, dist2_var ** 0.5)
+
+    res = (kl_divergence(dist1, dist2) + kl_divergence(dist2, dist1)).sum(-1)
+    return res.transpose(-1, -2)
+
+
+class DistributionalInputKernel(Kernel):
+    r"""
+    Computes a covariance matrix over __Gaussian__ distributions via exponentiating the
+    distance function between probability distributions.
+    .. math::
+
+        \begin{equation*}
+            k(p(x), p(x')) = \exp\{-a d(p(x), p(x'))\})
+        \end{equation*}
+
+    where :math:`a` is the lengthscale.
+    The implemented distance functions comes from Moreno et al, '04
+    (https://papers.nips.cc/paper/2351-a-kullback-leibler-divergence-based-kernel-for-svm-\
+    classification-in-multimedia-applications.pdf) for the symmetrized KL. For
+    general distributions, these kernels may not produce PSD kernels; however, they do for
+    Gaussian distributions which are implemented here.
+
+    Args:
+        :attr:`distance` (string, optional, default=`symmetrized_kl`):
+            `symmetrized_kl`. Determines whether to use the
+            exponentiated symmetrized KL divergence.
+        :attr:`num_dims` (int): The first `num_dims` dimensions of the last dimension are
```

Comment:
seems unnecessary to specify this - see comment above

Reply:
My original plan for the PR was to support more than just the symmetrized_KL but I couldn't immediately verify that the Wasserstein implementation was behaving properly.

I'll remove for now.